tests/util/bft.py

```diff
@@ -290,19 +290,59 @@ def get_live_replicas(self):
             expected = lambda _: True
 
         try:
+            # Step 1: Wait for an agreed view that matches the "expected" predicate
```

Comment:
I get what you are trying to do here with the use of closures, but the nesting makes it hard to see what the loops contain. Can you please break this up into non-nested named functions?